plugins/source/azure/resources/services/monitor/subscription_diagnostic_settings.go

```diff
@@ -0,0 +1,40 @@
+package monitor
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/monitor/armmonitor"
+	"github.com/cloudquery/cloudquery/plugins/source/azure/client"
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/transformers"
+)
+
+func SubscriptionDiagnosticSettings() *schema.Table {
+	return &schema.Table{
+		Name:        "azure_monitor_subscription_diagnostic_settings",
+		Resolver:    fetchSubscriptionDiagnosticSettings,
+		Description: "https://learn.microsoft.com/en-us/rest/api/monitor/subscription-diagnostic-settings/list?tabs=HTTP#subscriptiondiagnosticsettingsresource",
+		Multiplex:   client.SubscriptionMultiplexRegisteredNamespace("azure_monitor_subscription_diagnostic_settings", client.Namespacemicrosoft_insights),
+		Transform:   transformers.TransformWithStruct(&armmonitor.DiagnosticSettingsResource{}, transformers.WithPrimaryKeys("ID")),
+		Columns:     schema.ColumnList{client.SubscriptionID},
```

Comment:
if `id` is the subscription id already, what's the point of adding `subscription_id`? `azure_subscription_subscriptions` is in a similar situation but it doesn't have the id duplication, so we can also give it a miss here?

Reply:
Let me verify but I think the ID is of the diagnostic setting and not the subscription

Reply:
![Screenshot 2023-03-06 at 11 30 51](https://user-images.githubusercontent.com/26760571/223071018-0f7da29a-1de2-4804-b631-f2282deb288f.png)

Yes, the ID is of the diagnostic setting